### ISSUE TITLE
Implement FT-Transformer Encoder module

### ISSUE THREAD

**Issue by takeshimx** [takeshimx/diamond-lens#76]
## Summary
Create `backend/app/services/ft_transformer.py` — the core FT-Transformer encoder with self-supervised training.

## Architecture
```
Input features (ops, iso, k_rate, bb_rate)
  → Feature Tokenizer (Linear per feature → d_model dim)
  → [CLS] token prepend
  → Transformer Encoder (Self-Attention)
  → [CLS] output → Projection → embedding vector
  → Decoder (reconstruction head, training only)
```

## Key Components
- [ ] `FTTransformerEncoder(nn.Module)` class
  - `feature_tokenizers`: `nn.ModuleList` of `nn.Linear(1, d_model)` per feature
  - `cls_token`: learnable `nn.Parameter`
  - `transformer`: `nn.TransformerEncoder` with `batch_first=True`
  - `projection`: `nn.Linear(d_model, embedding_dim)`
  - `decoder`: `nn.Linear(embedding_dim, n_features)` for reconstruction
- [ ] `encode(x)` method → returns `(batch, embedding_dim)` embeddings
- [ ] `forward(x)` method → returns `(embedding, reconstructed)` tuple
- [ ] `train_ft_transformer()` function
  - StandardScaler normalization (for gradient stability)
  - MSE reconstruction loss
  - Adam optimizer, configurable epochs/lr/batch_size
  - Returns `(model, scaler, embeddings)`

## Hyperparameters (defaults)
| Param | Value | Rationale |
|-------|-------|-----------|
| d_model | 32 | Small feature set (3-4 features) |
| n_heads | 4 | head_dim = 32/4 = 8 |
| n_layers | 2 | Prevent overfitting on ~150 samples |
| embedding_dim | 16 | Sufficient expressiveness for K-means |
| epochs | 200 | Converges on small data |
| lr | 1e-3 | Adam default |

## Why self-supervised?
No labels exist for clustering. The reconstruction task (autoencoder style) forces the model to learn compressed representations that capture feature interactions via attention.

## Files
- `backend/app/services/ft_transformer.py` (NEW)

**Comment by takeshimx** [comment]
Consolidated into fewer issues with better granularity.
